### ISSUE TITLE
focus style covers up pressed style

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot#42322]
**Godot version:**
3.2.3

**OS/device including version:**
GLES3 possibly 2 as well

**Issue description:**
the focus style covers up the pressed style making some gui styles unable to be made

**Steps to reproduce:**
- download project
- run Debug.tscn
- click on button
- realize the focus of the button covers the pressed style

**Minimal reproduction project:**
[FocusBug.zip](https://github.com/godotengine/godot/files/5279795/FocusBug.zip)

**Notes:**
I have brought up this issue before and it was shut down from people misunderstanding the issue
basically styles for controls should be layered like the following..
- normal
- focus
- hover
- pressed
- disabled

this assures that people can use transparent styles if needed or opaque styles if needed.
right now its impossible to use opaque focus styles because it covers up everything.

this makes things like a win95 theme impossible

**Comment by Calinou** [comment]
I believe this is done by design. This way, the focus style is still displayed if you press the button (and doesn't magically disappear when you press it).

**Comment by Shadowblitz16** [comment]
I honestly think there should be two focus styles.
one for the focus overlay and one for the actual control style

the win32 api does this...
![image](https://user-images.githubusercontent.com/2453707/94297071-c2c1cc00-ff18-11ea-8548-5bba0f418fa8.png)
the first one is the normal state, the second one is the focus state, and the third one is the pressed state.
however you can see that I have a focus rectangle drawn over the pressed and focused styles

**Comment by theoway** [comment]
@Calinou Where should I mention this in the docs? In the `StyleBox` or `Control` class reference?

**Comment by Calinou** [comment]
@theoway Probably both for completeness' sake :slightly_smiling_face:

**Comment by theoway** [comment]
> I honestly think there should be two focus styles.
> one for the focus overlay and one for the actual control style
> 
> the win32 api does this...
> ![image](https://user-images.githubusercontent.com/2453707/94297071-c2c1cc00-ff18-11ea-8548-5bba0f418fa8.png)
> the first one is the normal state, the second one is the focus state, and the third one is the pressed state.
> however you can see that I have a focus rectangle drawn over the pressed and focused styles
> its a hack and is not ideal but its the best I can do in godot right now

I've been searching, but couldn't understand the difference between `hover` & `focus`

**Comment by Shadowblitz16** [comment]
@akien-mga  This doesn't fix the issue though!
what are we suppose to use for a actual focus style then?

what your using it for is more for a focus overlay not a style

I feel like your ignoring this issue
